### DISCUSSION TITLE
Cara menambahkan nama jalan di map (leaflet) R Studio

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#611] [Q&A] [answered]
Saya punya kode seperti ini : 

latitude <- 22.3193
longitude <-  114.1690
map <- leaflet() %>%
    addTiles() %>%
    addMarkers(lng=longitude, lat=latitude)

Bagaimana cara menampilkan nama jalan dari lokasi latlongnya ketika pointnya diklik ?

Terima kasih sblmnya 🙏

**Comment by Yusraf002** [comment]
Dear @yohanespm77 

Terima kasih atas pertanyaannya pak, terkait pertanyaan Bapak untuk menambahkan label berupa nama jalan. Bapak bisa menggunakan function `addPopups()`, untuk langkah-langkah detail Bapak bisa mengakses dan mempelajari lebih lanjut melalui link berikut [Leaflet for R](https://rstudio.github.io/leaflet/popups.html).

Sekian jawaban dari kami, jika jawaban ini dirasa membantu bisa klik "Mark as Answer" pada comment ini.

Terima kasih,
Team Algoritma

**Reply by yohanespm77** [reply]
Maaf kak, saya ingin menambahkan nama jalannya berdasarkan latitude dan longitudenya (dinamis) kak, bukan diinput sendiri. Sebab jika diinput sendiri, ketika latitude dan longitudenya berubah, nama jalannya tidak berubah (statis).

**Reply by Yusraf002** [reply]
Dear @yohanespm77

untuk menambahkan label berupa nama jalan pada function `addMarkers` selain memasukkan parameter `lng` dan `lat`, Bapak bisa menambahkan parameter `popup`. seperti yang tercantum pada link berikut 
[Customize Leaflet Maps](https://www.earthdatascience.org/courses/earth-analytics/get-data-using-apis/leaflet-r/)

Sekian jawaban dari kami, jika jawaban ini dirasa membantu bisa klik "Mark as Answer" pada comment ini.

Terima kasih,
Team Algoritma

**Reply by yohanespm77** [reply]
Maaf kak, bisa diberikan contoh codenya ? Sebab referensi yang kakak berikan sepertinya tidak menjawab pertanyaan saya kak 🙏

Saya ingin nama jalannya muncul sesuai dengan latlong yg diberikan.

**Comment by IrfanChairurrachman** [comment, marked as answer]
Dear @yohanespm77 ,

Untuk mendapatkan informasi lokasi hanya berdasarkan latitude dan longitude harus menggunakan sumber external, dalam hal ini adalah osm atw `openstreetmap`. Berikut kami berikan contoh kode penerapan openstreetmap untuk mendapatkan informasi lokasi:

```{r}
library(leaflet)
library(jsonlite)
library(leaflet.providers)
# Mendefinisikan latitude dan longitude
latitude <- -6.2672131395077555
longitude <- 107.11880197831752
# Membuat map
map <- leaflet() %>%
  addProviderTiles(providers$OpenStreetMap.Mapnik) %>%
  setView(lng = longitude, lat = latitude, zoom = 16)
# Menambahkan layer geojson dari OpenStreetMap
geojson_url <- paste0("https://nominatim.openstreetmap.org/reverse?format=geojson&lat=",
                      latitude, "&lon=", longitude)
jalan_geojson <- jsonlite::fromJSON(geojson_url)
map %>% addTopoJSON(jalan_geojson)
# Menambahkan marker dan nama jalan pada popup
map %>% addMarkers(lng = longitude, lat = latitude,
                   popup = jalan_geojson$features$properties$display_name)
```

Pada bagian popup-nya, Anda bisa memilih informasi yang ingin ditampilkan. Pilihan informasi berupa kolom dataframe yang dapat Anda akses dengan cara `chaining` seperti berikut: `jalan_geojson$features$properties`.

Sekian informasi dari kami, jika dirasa membantu dapat klik "mark as answer" pada comment ini. Terima kasih

Best Regards,
Algoritma Team.

**Reply by yohanespm77** [reply]
Terima kasih atas jawabannya kak. Sangat membantu 🙏
